contrib/mesos/pkg/executor/executor.go

```diff
@@ -1044,3 +1002,220 @@ func executorInfoToAnnotations(ei *mesos.ExecutorInfo) (annotations map[string]s
 
 	return
 }
+
+// implements txTaskOperator, uses the executor state lock to handle concurrent invocation
+func (k *Executor) doForTask(taskId string, f func(*kuberTask, *api.Pod) bool) (changed bool, err error) {
```

Comment:
Now I get it: you are implementing a transaction pattern. Why don't you call it `modifyTask(taskId, modifier)`